app/models/user_preference.rb

```diff
@@ -3,8 +3,12 @@ class UserPreference
 
   def initialize
     array = []
-    Metacode.all.find_each do |m|
-      array.push(m.id.to_s)
+    begin
+      Metacode.all.find_each do |m|
+        array.push(m.id.to_s)
+      end
+    rescue ActiveRecord::StatementInvalid
+      Rails.logger.warn("TODO: remove this travis workaround in user_preference.rb")
```

Comment:
I cant recall precisely when the error happens, so it may be just an "cant
initialize an empty database" workaround, but when you try to load the
schema it fails because this class tries to initialize and it cant find the
Metacode table.

It definitely happens on every travis run, and it might also happen for
people initializing a local db.

On Tue, Sep 13, 2016, 11:22 AM Harlan T Wood <notifications@github.com>
wrote:

> In app/models/user_preference.rb
> <https://github.com/metamaps/metamaps/pull/593#discussion_r78491248>:
>
> > @@ -3,8 +3,12 @@ class UserPreference
> >
> >    def initialize
> >      array = []
> > -    Metacode.all.find_each do |m|
> > -      array.push(m.id.to_s)
> > +    begin
> > +      Metacode.all.find_each do |m|
> > +        array.push(m.id.to_s)
> > +      end
> > +    rescue ActiveRecord::StatementInvalid
> > +      Rails.logger.warn("TODO: remove this travis workaround in user_preference.rb")
>
> In what sense is this a Travis workaround? Curious.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/metamaps/metamaps/pull/593/files/f4ac81b0b71e676d59fb6f3f2c8a14d9dea0f62b..1c38865db3d823e7da9686d701aaff8265236e20#r78491248>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABVELNjgm-B_BKj07AGQqYRz836th3i3ks5qphb0gaJpZM4JeexP>
> .
>